power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/sparse_lu_solver.hpp

```diff
@@ -21,30 +21,30 @@ template <class Tensor, class RHSVector, class XVector, class = void>
 struct sparse_lu_entry_trait;
 
 template <class Tensor, class RHSVector, class XVector>
-using enable_scalar_lu_t =
-    std::enable_if_t<std::is_same_v<Tensor, RHSVector> && std::is_same_v<Tensor, XVector> && check_scalar_v<Tensor>>;
+concept scalar_value_lu = scalar_value<Tensor> && std::same_as<Tensor, RHSVector> && std::same_as<Tensor, XVector>;
 
 template <class Derived>
 int check_array_base(Eigen::ArrayBase<Derived> const&) {
     return 0;
 }
 
+template <class ArrayLike>
+concept eigen_array = std::same_as<decltype(check_array_base(ArrayLike{})), int>;  // should be an eigen array
```

Comment:
This is a concept which will be ill-formed for scalar. It is still `enable_if` like approach. Do we have a better idea to do it here?

Reply:
hmmm i agree that this is not a good implementation. you want to test iff it expands to something of the template `Eigen::ArrayBase<T>`, correct? let me think about it for a bit. I remember doing something similar when I was at Philips.

If you prefer, we can also decide to do that in a separate PR. In the meantime, I will experiment with this a bit

Reply:
Yes, that's out of scope here. Let's keep it for another time.